pkg/ebpf/bpf_error.go

```diff
@@ -0,0 +1,194 @@
+package ebpf
+
+import (
+	"encoding/binary"
+	"fmt"
+	"strings"
+	"unsafe"
+
+	"github.com/aquasecurity/tracee/pkg/logger"
+)
+
+const BPFMaxErrFileLen = 72 // BPF_MAX_ERR_FILE_LEN
+
+type BPFErrorType uint32
+
+const (
+	BPFErrUnspec BPFErrorType = iota // BPF_ERR_UNSPEC
+
+	BPFErrInitContext // BPF_ERR_INIT_CONTEXT
+
+	BPFErrMapUpdateElem  // BPF_ERR_MAP_UPDATE_ELEM
+	BPFErrMapDeleteElem  // BPF_ERR_MAP_DELETE_ELEM
+	BPFErrGetCurrentComm // BPF_ERR_GET_CURRENT_COMM
+	BPFErrTailCall       // BPF_ERR_TAIL_CALL
+)
+
+var stringMap = map[BPFErrorType]string{
+	BPFErrUnspec: "BPF_ERR_UNSPEC",
+
+	BPFErrInitContext: "BPF_ERR_INIT_CONTEXT",
+
+	BPFErrMapUpdateElem:  "BPF_ERR_MAP_UPDATE_ELEM",
+	BPFErrMapDeleteElem:  "BPF_ERR_MAP_DELETE_ELEM",
+	BPFErrGetCurrentComm: "BPF_ERR_GET_CURRENT_COMM",
+	BPFErrTailCall:       "BPF_ERR_TAIL_CALL",
+}
+
+var errorMap = map[BPFErrorType]string{
+	BPFErrUnspec: "Unspecifed BPF error",
+
+	BPFErrInitContext: "Failed to init context",
+
+	BPFErrMapUpdateElem:  "Failed to add or update a map element",
+	BPFErrMapDeleteElem:  "Failed to delete a map element",
+	BPFErrGetCurrentComm: "Failed to get current command",
+	BPFErrTailCall:       "Failed to tail call",
+}
+
+func (b BPFErrorType) String() string {
+	if bpfErr, found := stringMap[b]; found {
+		return bpfErr
+	}
+
+	return stringMap[BPFErrUnspec]
+}
+
+type BPFErrorLogLevel int32
+
+const (
+	BPFErrLogLvlDebug BPFErrorLogLevel = iota + BPFErrorLogLevel(logger.DebugLevel) // BPF_ERR_LOG_LVL_DEBUG
```

Comment:
This (and the threesubsequent constants) are unused and as such should be removed.

Reply:
Ok. I got it.